test/functional/helper/AccessTokenRequester.groovy

```diff
@@ -16,7 +16,14 @@ class AccessTokenRequester {
     }
 
     static String getAccessToken(Map params) {
-        def response = requestAccessToken(params)
+        params = params.clone() // Don't mess with the original Map
+        String clientId = params.remove('client_id')
+        String clientSecret = params.remove('client_secret') ?: ''
+
+        def response = clientId != null ?
```

Comment:
This conditional isn't necessary. I applied the patch locally and confirmed this. 

Additionally, the `requestAccessToken` method (which returns the entire response) should be updated to also use basic auth for its request. I did this locally and had 13 tests fail due to the `Set-Cookie` header being present in the response. The `NullRememberMeServices` should resolve this.

My code for reference:

```groovy
    static HttpResponseDecorator requestAccessToken(Map requestParams) {
        Map params = requestParams.clone() as Map // Don't mess with the original Map
        String clientId = params.remove('client_id')
        String clientSecret = params.remove('client_secret') ?: ''

        requestAccessTokenWithBasicAuth(params, clientId, clientSecret)
    }

    static String getAccessToken(Map params) {
        def response = requestAccessToken(params)
        return response.data.access_token
    }
```